src/systems/movement/toggle_walk_run.rs

```diff
@@ -1,19 +1,17 @@
 use crate::{behavior::MovementMode, components::Player};
 use bevy::prelude::*;
 
-pub fn toggle_walk_run(mut player: Query<&mut Player>, keys: Res<Input<KeyCode>>) {
-	let Ok(mut player) = player.get_single_mut() else {
-		return; //FIXME: handle properly
-	};
-
+pub fn player_toggle_walk_run(mut player: Query<&mut Player>, keys: Res<Input<KeyCode>>) {
 	if !keys.just_pressed(KeyCode::NumpadSubtract) {
 		return;
 	}
 
-	player.movement_mode = match player.movement_mode {
-		MovementMode::Run => MovementMode::Walk,
-		MovementMode::Walk => MovementMode::Run,
-	};
+	for mut player in player.iter_mut() {
+		player.movement_mode = match player.movement_mode {
+			MovementMode::Walk => MovementMode::Run,
+			MovementMode::Run => MovementMode::Walk,
+		}
+	}
```

Comment:
Lol, I changed that like a million times over and ended up with the same implementation as the first time...

Reply:
I know, I was there. Talk to yourself much?